### PR TITLE
[DNM] Upmerge 04 11 2019

### PR DESCRIPTION
**Pull request by de-nordic**

### CODE REVIEW

**Comment by mbolivar** [comment]
@de-nordic for your reference, here is a rebased tree for this mergeup with the same code:

https://github.com/mbolivar/fw-nrfconnect-zephyr/tree/mergeup-231-rebase

I just had to skip cherry-picking 37ef787. No other issues.

Nice job!

**Comment by de-nordic** [comment]
> 
> 
> @de-nordic for your reference, here is a rebased tree for this mergeup with the same code:
> 
> https://github.com/mbolivar/fw-nrfconnect-zephyr/tree/mergeup-231-rebase
> 
> I just had to skip cherry-picking [37ef787](https://github.com/NordicPlayground/fw-nrfconnect-zephyr/commit/37ef7876836b36ecd3d9e21c79c85e4cdcffec04). No other issues.
> 
> Nice job!

I have done the requested change, It was not much of an effort. Thanks for pointing out the duplicate.

File: cmake/dts.cmake [hunk 1]
```diff
@@ -16,8 +16,12 @@ set(GENERATED_DTS_BOARD_UNFIXED_H ${PROJECT_BINARY_DIR}/include/generated/genera
 set(GENERATED_DTS_BOARD_CONF      ${PROJECT_BINARY_DIR}/include/generated/generated_dts_board.conf)
 set(DTS_POST_CPP                  ${PROJECT_BINARY_DIR}/${BOARD}.dts.pre.tmp)
 
-set_ifndef(${IMAGE}DTS_SOURCE ${BOARD_DIR}/${BOARD}.dts)
-set_ifndef(${IMAGE}DTS_COMMON_OVERLAYS ${ZEPHYR_BASE}/dts/common/common.dts)
+set_ifndef(DTS_SOURCE ${BOARD_DIR}/${BOARD}.dts)
```
Comment (sigvartmh) [inline cmake/dts.cmake]:
```suggestion
set_ifndef(${IMAGE}DTS_SOURCE ${BOARD_DIR}/${BOARD}.dts)
```

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -143,6 +143,7 @@ comment "BLE Controller configuration"
 config BT_CTLR_CRYPTO
 	bool "Enable crypto functions in Controller"
 	default y
+	select ENTROPY_GENERATOR
```
Comment (anangl) [inline subsys/bluetooth/controller/Kconfig]: The commit containing this change is tagged `[nrf noup]`. Shouldn't it actually go upstream? Enabling `BT_CTLR_CRYPTO` [causes](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/CMakeLists.txt#L17) `subsys/bluetooth/controller/crypto/crypto.c` to be compiled and this file [expects](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/crypto/crypto.c#L26) an entropy generator to be available.

File: cmake/dts.cmake (hunk 1 above)
Comment (de-nordic) [reply cmake/dts.cmake]: Thanks!

**Comment by tejlmand** [comment]
@de-nordic @mbolivar Did someone forget to tag before merging this PR ?
If I look at this manifest version: https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/4b5a62558213f5a080bc366535f9ecab041c6594/west.yml#L41
it points to the following SHA:
9be97c63368aa5ddfb7613446794a3ec85801a49
but I don't see that SHA included in any tag, meaning that users who don't have it already might not be able to check out this revision.
```
ncs/zephyr$ git tag --contains 9be97c63368aa5ddfb7613446794a3ec85801a49
```
returns nothing.

**Comment by tejlmand** [comment]
@mbolivar just discovered, the reason it's not in any tag, is because it hasn't been tagged yet.
But the commit is still there in master, just far down in the git log :)

**Comment by mbolivar** [comment]
> @mbolivar just discovered, the reason it's not in any tag, is because it hasn't been tagged yet.
> But the commit is still there in master, just far down in the git log :)

Phew! :)

**Pull request merged**